router/internal/persistedoperation/cache.go

```diff
@@ -0,0 +1,31 @@
+package persistedoperation
+
+import (
+	"github.com/dgraph-io/ristretto"
+	"github.com/wundergraph/cosmo/router/internal/unsafebytes"
+)
+
+const (
+	persistentAverageCacheEntrySize = 4 * 1024 // 4kb
+)
+
+type OperationsCache struct {
+	// cache is the backing store for the in-memory cache. Note
+	// that if the cache is disabled, this will be nil
+	Cache *ristretto.Cache[string, []byte]
+}
+
+func (c *OperationsCache) key(clientName string, operationHash []byte) string {
+	return clientName + unsafebytes.BytesToString(operationHash)
+}
+
+func (c *OperationsCache) Get(clientName string, operationHash string) []byte {
+	// Since we're returning nil when the item is not found, we don't need to
+	// check the return value from the cache nor the type assertion
+	item, _ := c.Cache.Get(c.key(clientName, unsafebytes.StringToBytes(operationHash)))
+	return item
+}
+
+func (c *OperationsCache) Set(clientName, operationHash string, operationBody []byte) {
+	c.Cache.Set(c.key(clientName, unsafebytes.StringToBytes(operationHash)), operationBody, int64(len(operationBody)))
```

Comment:
I don't think it's a good idea to use an unsafe string as key.
Use xxhash and uint64 instead, which is safe and fast.

Reply:
Fixed, byte conversion was actually not necessary